crates/templates/src/template.rs

```diff
@@ -0,0 +1,164 @@
+                manifest_path.display()
+            )
+        })?;
+        let raw = crate::reader::parse_manifest_toml(&manifest_text).with_context(|| {
+            format!(
+                "Manifest file {} is not a valid manifest",
+                manifest_path.display()
+            )
+        })?;
+
+        let content_dir = if layout.content_dir().exists() {
+            Some(layout.content_dir())
+        } else {
+            None
+        };
+
+        let template = match raw {
+            RawTemplateManifest::V1(raw) => Self {
+                id: raw.id.clone(),
+                description: raw.description.clone(),
+                parameters: Self::parse_parameters(&raw.parameters)?,
+                content_dir,
+            },
+        };
+        Ok(template)
+    }
+
+    pub fn id(&self) -> String {
+        self.id.clone()
+    }
+
+    pub fn description(&self) -> Option<String> {
+        self.description.clone()
+    }
+
+    pub(crate) fn parameters(&self) -> impl Iterator<Item = &TemplateParameter> {
+        self.parameters.iter()
+    }
+
+    pub(crate) fn parameter(&self, name: impl AsRef<str>) -> Option<&TemplateParameter> {
+        self.parameters.iter().find(|p| p.id == name.as_ref())
+    }
+
+    pub(crate) fn content_dir(&self) -> Option<PathBuf> {
+        self.content_dir.clone()
+    }
+
+    pub fn run(self, options: RunOptions) -> Run {
+        Run::new(self, options)
+    }
+
+    fn parse_parameters(
+        raw: &Option<IndexMap<String, RawParameter>>,
+    ) -> anyhow::Result<Vec<TemplateParameter>> {
+        match raw {
+            None => Ok(vec![]),
+            Some(parameters) => parameters
+                .iter()
+                .map(|(k, v)| TemplateParameter::from_raw(k, v))
+                .collect(),
+        }
+    }
+}
+
+impl TemplateParameter {
+    fn from_raw(id: &str, raw: &RawParameter) -> anyhow::Result<Self> {
+        let data_type = TemplateParameterDataType::parse(raw)?;
+
+        Ok(Self {
+            id: id.to_owned(),
+            data_type,
+            prompt: raw.prompt.clone(),
+            default_value: raw.default_value.clone(),
+        })
+    }
+
+    pub fn id(&self) -> String {
+        self.id.clone()
+    }
+
+    pub fn data_type(&self) -> TemplateParameterDataType {
+        self.data_type.clone()
+    }
+
+    pub fn prompt(&self) -> String {
+        self.prompt.clone()
+    }
+
+    pub fn default_value(&self) -> Option<String> {
+        self.default_value.clone()
+    }
+
+    pub fn validate_value(&self, value: impl AsRef<str>) -> anyhow::Result<String> {
+        self.data_type.validate_value(value.as_ref().to_owned())
+    }
+}
+
+impl TemplateParameterDataType {
+    fn parse(raw: &RawParameter) -> anyhow::Result<Self> {
+        match &raw.data_type[..] {
+            "string" => Ok(Self::String(parse_string_constraints(raw)?)),
+            "number" => Ok(Self::Number),
+            "boolean" => Ok(Self::Boolean),
```

Comment:
If I'm interpretting correctly, it looks like we're allowing these types in the manifest, but then panicing at runtime? That doesn't seem ideal...

Reply:
Well, these were placeholders for Brian.  I can remove them and he can bring them back though - probably safer!